observer/lib/observer.js

```diff
@@ -35,8 +35,24 @@ export const observeTransferEvents = async (pgPoolStats, ieContract, provider) =
   return events.length
 }
 
+const getScheduledRewards = async (address, ieContract) => {
+  const [fromContract, fromSparkRewards] = await Promise.all([
+    ieContract.rewardsScheduledFor(address),
+    (async () => {
+      const res = await fetch(
+        `https://spark-rewards.fly.dev/scheduled-rewards/${address}`
+      )
+      const json = await res.json()
```

Comment:
How do you handle 4xx and 5xx responses? What if the server responds with 5xx and the response body provides error details? Should you check `res.ok`?

In other places, I had a great experience with the helper `assertOkResponse` that reads the error response body and includes it in `err.message`.

Reply:
see reply to same question in other PR

Reply:
Link to that reply: https://github.com/filecoin-station/core/pull/606#discussion_r1791853853